### PR TITLE
Add tests and implementation for intersectsExtent (ol.geom.Geometry)

### PR DESCRIPTION
**Pull request by alvinlindstam**
This PR implements ol.geom.Circle.prototype.intersectsExtent, and adds tests for that method on all geometry types that support it.

I noticed the missing tests and support on ol.geom.Circle while doing PR https://github.com/openlayers/ol3/pull/3761.

I didn't implement intersectsExtent on ol.geom.LinearRing since I noticed it's not supposed to be used as anything else than polygon components. After this PR, allt renderable geometry types support intersectsExtent, which would make it useable for ol.source.Vector#getFeaturesAtCoordinate in PR https://github.com/openlayers/ol3/pull/3761

The implementation for Circle#intersectsExtent treats it as a surface, so any extent intersecting it's contents or boundary will return true.

Feel free to comment, I'll fix and rebase if I've missed any important conventions.

### CODE REVIEW

File: src/ol/geom/circle.js [hunk 1]
```diff
@@ -138,6 +138,30 @@ ol.geom.Circle.prototype.getType = function() {
...
 
 /**
+ * @inheritDoc
+ * @api stable
+ */
+ol.geom.Circle.prototype.intersectsExtent = function(extent) {
+  var circleExtent = this.getExtent();
+  if (ol.extent.intersects(extent, circleExtent)) {
+    var centerX = this.flatCoordinates[0];
+    var centerY = this.flatCoordinates[0];
```
Comment (marcjansen) [inline src/ol/geom/circle.js]: Must be `this.flatCoordinates[1];`, right?

Comment (alvinlindstam) [inline src/ol/geom/circle.js]: Yes, of course. I've pushed an update

**Comment by marcjansen** [comment]
You may want to cherry-pick marcjansen/ol3@e06bb6cbfb2ec7195b5b8c374b750802664062cb.

Do we have a [CLA](https://github.com/openlayers/ol3/blob/master/CONTRIBUTING.md#contributor-license-agreement) from you?

**Comment by alvinlindstam** [comment]
I merged the commits just adding tests, and kept the separate commit for the Circle geom and your additional commit. Should i merge my commits into one?

Yes, I've submitted a private CLA and are contributing as an individual.

**Comment by marcjansen** [comment]
> Should i merge all my commits into one?

No, this looks fine now. Let's wait for CI

**Comment by marcjansen** [comment]
The commit message for 7e2bca6b8838cc5f84093b0825f4dab25e7e0214 is not correct. I suggest rewording to sth. like:

```
Add tests for intersectsExtent/getExtent

This commit adds tests for `intersectsExtent` of a, b, c,
and a basic test for `getExtent` of e.
```

**Comment by marcjansen** [comment]
Once this is done, I'll merge. Thanks for the work on this, @alvinlindstam.

**Comment by marcjansen** [comment]
Very nice, thanks.
